### PR TITLE
permissions: refactoring organisation policy

### PR DESCRIPTION
**Pull request by zannkukai**
* Refactoring Organisation policy access using `Permission` framework.
* Adapts `PatronType` policy.
* Adapts unitest

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

### CODE REVIEW

File: rero_ils/modules/ext.py [hunk 1]
```diff
@@ -71,6 +74,17 @@
 from ..version import __version__
 
 
+@identity_loaded.connect
+def on_identity_loaded(_, identity):
+    """Add ``Needs`` to manage RERO-ILS permissions."""
```
Comment (jma) [inline rero_ils/modules/ext.py:79]: The comment can be more detailed and should contains `@param` description.

File: rero_ils/modules/ext.py [hunk 2]
```diff
@@ -71,6 +74,17 @@
...
+    if current_librarian:
+        identity.provides.add(
```
Comment (jma) [inline rero_ils/modules/ext.py:81]: You can compute a list of organisation pids to add depending if it is a librarian or a patrons and add the needs at the end of the function.

File: rero_ils/modules/permissions.py [hunk 3]
```diff
@@ -359,31 +360,32 @@ class AllowedByAction(Generator):
...
 
-    def excludes(self, record=None, **kwargs):
-        """Excludes if the record and the user has different organisation.
+    def needs(self, record=None, *args, **kwargs):
+        """Allows the given action.
 
-        :param record: Record instance - the given record.
-        :returns: a list of permissions, empty if allows.
+        :param record: the record to check.
+        :param kwargs: extra arguments.
```
Comment (jma) [inline rero_ils/modules/permissions.py:383]: `*args` description is missing.

**Review by jma — APPROVED** [review]

File: rero_ils/modules/ext.py (hunk 2 above)
Comment (zannkukai) [reply rero_ils/modules/ext.py:81]: of course. I tried but readability is degraded.

**Review by lauren-d — APPROVED** [review]

**Pull request merged**